FidelityFusion_Models/AR_autoRegression.py

```diff
@@ -0,0 +1,133 @@
+import sys
+import os
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+import torch
+import torch.nn as nn
+import GaussianProcess.kernel as kernel
+from GaussianProcess.gp_basic import GP_basic as CIGP
+from MF_data import MultiFidelityDataManager
+import matplotlib.pyplot as plt
+
+class autoRegression(nn.Module):
+    # initialize the model
+    def __init__(self,fidelity_num,kernel,rho_init=1.0,nonsubset=False):
+        super().__init__()
+        self.fidelity_num = fidelity_num
+        # create the model
+        self.cigp_list=[]
+        for _ in range(self.fidelity_num):
+            self.cigp_list.append(CIGP(kernel=kernel, noise_variance=1.0))
+        
+        self.cigp_list = torch.nn.ModuleList(self.cigp_list)
+
+        self.rho_list=[]
+        for _ in range(self.fidelity_num-1):
+            self.rho_list.append(torch.nn.Parameter(torch.tensor(rho_init)))
+
+        self.rho_list = torch.nn.ParameterList(self.rho_list)
+        self.nonsubset = nonsubset
+
+    def forward(self,data_manager,x_test,to_fidelity=None):
+        # predict the model
+        if to_fidelity is not None and to_fidelity >= 1:
+            fidelity_num = to_fidelity
+        else:
+            fidelity_num = self.fidelity_num
+        for f in range(fidelity_num):
+            if f == 0:
+                x_train,y_train = data_manager.get_data(f)
+                y_pred_low, cov_pred_low = self.cigp_list[f](x_train,y_train,x_test)
+                if fidelity_num == 1:
+                    y_pred_high = y_pred_low
+                    cov_pred_high = cov_pred_low
+            else:
+                x_train,y_train = data_manager.get_data_by_name('res-{}'.format(f))
```

Comment:
left hand side y_pred_high does not have f but the right hand size has f, this can be confusing